mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -765,4 +771,37 @@ def AMDGPU_WMMAOp :
   let hasVerifier = 1;
 }
 
+def GlobalLoadMemRefType : MemRefOf<[GlobalLoadTypes]>;
+
+def AMDGPU_GlobalLoadLDSOp :
+    AMDGPU_Op<"global_load", [SameVariadicOperandSize]>,
+    Arguments<(ins
+                   Arg<GlobalLoadMemRefType, "buffer to read from", [MemRead]>:$src,
+                   Variadic<I32>:$srcIndices,
+                   Arg<GlobalLoadMemRefType, "buffer to write to", [MemWrite]>:$dst,
+                   Variadic<I32>:$dstIndices
```

Comment:
Same - `index` here, it'll get truncated down

Reply:
Are you sure? Other ops (like `RawBufferLoadOp`) are using `I32` at this low level so I assume we should use a machine type...